src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -393,50 +395,77 @@ public void Initialize()
             using (ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
             {
                 if (TryLoadAuthenticationKeyPairs(ctx) == false)
-                    GenerateAuthenticationKeyPairs(ctx);
+                    GenerateAuthenticationSignetureKeys(ctx);                
             }
+            GenerateAuthenticationBoxKeys();
 
             Task.Run(ClusterMaintenanceSetupTask, ServerShutdown);
         }
 
         public byte[] BoxPublicKey, BoxSecretKey, SignPublicKey, SignSecretKey;
 
-        private void GenerateAuthenticationKeyPairs(TransactionOperationContext ctx)
+        private Tuple<DateTime, string> _clusterToken = Tuple.Create<DateTime, string>(DateTime.MinValue, null);
+
+        internal string GetClusterTokenForNode(JsonOperationContext context)
+        {
+            if (DateTime.UtcNow > _clusterToken.Item1 || _clusterToken.Item2 == null)
```

Comment:
i was thinking about this point yesterday, was thinking to regenerate them if the time passed is 20 min so we will have overlapping 10 minutes to handle this.
Redirect to leader use this so it isn't that rare as one would think.